presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -200,20 +200,29 @@ private static DwrfSequenceEncoding toSequenceEncoding(OrcType type, DwrfProto.C
                         columnEncoding.getDictionarySize()));
     }
 
-    private static Optional<SortedMap<Integer, DwrfSequenceEncoding>> toAdditionalSequenceEncodings(List<DwrfProto.ColumnEncoding> columnEncodings, OrcType type)
+    private static ColumnEncoding toColumnEncoding(OrcType type, List<DwrfProto.ColumnEncoding> columnEncodings)
     {
-        if (columnEncodings.size() == 1) {
-            return Optional.empty();
+        DwrfProto.ColumnEncoding sequence0 = null;
+        ImmutableSortedMap.Builder<Integer, DwrfSequenceEncoding> builder = ImmutableSortedMap.naturalOrder();
+        for (DwrfProto.ColumnEncoding columnEncoding : columnEncodings) {
+            if (columnEncoding.getSequence() == 0) {
+                sequence0 = columnEncoding;
+            }
+            else {
+                builder.put(columnEncoding.getSequence(), toSequenceEncoding(type, columnEncoding));
+            }
         }
 
-        ImmutableSortedMap.Builder<Integer, DwrfSequenceEncoding> additionalSequenceEncodings = ImmutableSortedMap.<Integer, DwrfSequenceEncoding>naturalOrder();
-
-        for (int i = 1; i < columnEncodings.size(); i++) {
-            DwrfProto.ColumnEncoding columnEncoding = columnEncodings.get(i);
-            additionalSequenceEncodings.put(columnEncoding.getSequence(), toSequenceEncoding(type, columnEncoding));
+        SortedMap<Integer, DwrfSequenceEncoding> nonZeroSequences = builder.build();
+        Optional<SortedMap<Integer, DwrfSequenceEncoding>> additionalSequenceEncodings =
```

Comment:
additionalSequenceEncodings name is confusing; the value is just nonZeroSequences wrapped into an Optional; Is it important to use Optional.empty() when nonZeroSequences is empty? Why can't we always use Optional.of(nonZeroSequences)? If that's possible, then additionalSequenceEncodings variable can be removed and Optional.of(nonZeroSequences) can be used in its place.

Reply:
I'm not sure if Optional.empty() is necessary. From looking at the code, it doesn't matter for now. However, grep told me there are places where it relies on Optional.isPresent(), so I attempted to maintain the existing behavior (ie. in the common case when there is only sequence 0 for a node, make additionalSequenceEncodings Optional.empty())